examples/summarization/model_ortds.py

```diff
@@ -0,0 +1,122 @@
+from typing import List, Dict
+import os
+
+# too long import
+from pymarlin.utils.stats import global_stats
+from pymarlin.utils.logger import getlogger
+
+from onnxruntime.training.ortmodule import ORTModule
+
+from filelock import FileLock
+
+from deepspeed_methods.deepspeed_utils import initialize_deepspeed
+from deepspeed_methods.deepspeed_utils import get_core_model
+
+from data import SummarizationData
+from train import SummarizationBartModule
+
+logger = getlogger(__file__)
+
+try:
+    import nltk
+
+    NLTK_AVAILABLE = True
+except (ImportError, ModuleNotFoundError):
+    NLTK_AVAILABLE = False
+
+if NLTK_AVAILABLE:
+    with FileLock(".lock") as lock:
+        nltk.download("punkt", quiet=True)
+
+
+class SummarizationBartModule_ds_ort(SummarizationBartModule):
+    def __init__(
+            self,
+            data: SummarizationData,
+            max_length_encoder=128,
+            max_length_decoder=128,
+            max_lr=2e-5,
+            ort=False,
+            deepspeed=False,
+            deepspeed_config='',
+            deepspeed_transformer_kernel=False,
+            deepspeed_ckpt_tag=None,
+            deepspeed_resume_from_checkpoint=None,
+            generate_kwargs={}
+    ):
+        super().__init__(data, max_length_encoder, max_length_decoder, max_lr, generate_kwargs)
+        
+        #setting this here to avoid issues after wrapping
+        self._pad_token_id = self.model.config.pad_token_id
+
+        if ort:
+            logger.info(f"Employing ORT, wrapping model with ORTModule")
+            self.model = ORTModule(self.model)
+        if deepspeed:
+            logger.info(f"Employing Deepspeed, wrapping model with Deepspeed")
+            self.model, _ = initialize_deepspeed(self.model, deepspeed_config, deepspeed_transformer_kernel)
+
+        self.ort = ort
+        self.deepspeed = deepspeed
+        self.deepspeed_resume_from_checkpoint = deepspeed_resume_from_checkpoint
+        self.deepspeed_ckpt_tag = deepspeed_ckpt_tag
+        self.DEEPSPEED_CKPT_PREFIX = "deepspeed_ckpt"
+    
+    @property
+    def pad_token_id(self):
+        return self._pad_token_id
+
+    def get_optimizers_schedulers(
+            self, estimated_global_steps_per_epoch: int, epochs: int
+    ):
+        if self.deepspeed:
```

Comment:
too many if statements for deepspeed. Possible refactorization using decorator pattern or multiple derived classes?

Reply:
I can make a seperate module interface that can extend for deepspeed, and a decorator/wrapper class for ORT.

Reply:
I cleaned these up and created an ORT and DeepSpeed module (DeepSpeed inherits from ORT for simplicity - but we could make these wrap for common trainers in the future).